### ISSUE TITLE
Procedure that doesn't yield results terminates query

### ISSUE THREAD

**Issue by jameslaneconkling** [neo4j-contrib/neo4j-apoc-procedures#1172]
A procedure that doesn't yield results terminates the query early in cases when addition data can be returned

## Expected Behavior (Mandatory)
```cypher
WITH [1,2,3] as a
CALL apoc.algo.cover([123, 456]) yield rel
RETURN a, collect(rel) as rels
// [{ a: [1,2,3], rels: [] }]
```

## Actual Behavior (Mandatory)
```cypher
WITH [1,2,3] as a
CALL apoc.algo.cover([123, 456]) yield rel
RETURN a, collect(rel) as rels
// []
```

If the function does yield data, the result is as expected
```cypher
WITH [1,2,3] as a
CALL apoc.algo.cover([62405344, 62405345]) yield rel
RETURN a, collect(rel) as rels
// [{ a: [1,2,3], rels: [<relationship>] }]
```

Interestingly, returning the collected empty procedure result _without_ other data does return an empty list
```cypher
WITH [1,2,3] as a
CALL apoc.algo.cover([123, 456]) yield rel
RETURN collect(rel) as rels
// [{ rels: [] }]
```

I've seen this happening with both `cover` and `subGraphNodes`, so I assume it's inherent to all functions.  

### Versions

  - OS: 10.14.4
  - Neo4j: 3.5.3
  - Neo4j-Apoc: 3.5.0.2

**Comment by jexp** [comment]
Hi James,

that's expected behavior. 

Any operation in Cypher that can generate 0..n rows 

Will have the query not generate results if it returned 0 rows.

So until we have real subqueries in cypher that is hard to resolve.

What behaivor do you expect?

I think the last query you shared has a copy & paste error.

There is one special case with standalone aggregations in Neo4j

Which return an empty default (0 or empty list) if there are **no aggregation keys* *

**Comment by jameslaneconkling** [comment]
> What behavior do you expect?

For cases where only the result of the procedure is returned, terminating the query and returning nothing is fine.  But for cases where other fields are returned (e.g. `a` in the above example), it's still important to not drop those.

If the query used `MATCH ...` instead of `CALL apoc...`, this can be handled via `OPTIONAL MATCH` (if I understand correctly).  But sounds like there's no equivalent for `CALL`.  Though yes, I imagine subquery support would also solve this.

I did find a workaround, though, using `apoc.cypher.run()`:
```cypher
WITH [1,2,3] as a
CALL apoc.cypher.run("CALL apoc.algo.cover([62405344, 123]) yield rel RETURN collect(rel) as rels", {}) yield value
RETURN a, value.rels
// [{ a: [1,2,3], rels: [] }]
```

> I think the last query you shared has a copy & paste error

Oops, yeah fixed.

Thanks for the pointers.  The workaround should cover the general use case.  Closing.
